pkg/network/vmispec/hotplug.go

```diff
@@ -69,3 +69,15 @@ func NetworksToHotplugWhosePodIfacesAreReady(vmi *v1.VirtualMachineInstance) []v
 
 	return networksToHotplug
 }
+
+func FilterIfacesForHotplug(ifaces []v1.Interface) []v1.Interface {
+	return FilterInterfacesSpec(ifaces, func(iface v1.Interface) bool {
+		return iface.State != v1.InterfaceStateAbsent && iface.SRIOV == nil
+	})
+}
+
+func FilterIfacesForHotUnplug(ifaces []v1.Interface) []v1.Interface {
+	return FilterInterfacesSpec(ifaces, func(iface v1.Interface) bool {
+		return iface.State == v1.InterfaceStateAbsent && iface.SRIOV == nil
+	})
+}
```

Comment:
Placing these helpers here is confusing. We will/do support hot-plug/unplug for SR-IOV, but one that reads this may assume we do not.

I think we can use the generic filter function where it is needed, because at that location SR-IOV is not treated. It will also remove the tests and unneeded boilerplate which comes with these.

Reply:
These are mere wrappers for the logic needed to get the interfaces to un/plug, with no boilerplate whatsoever.

Note that having these functions enable testing the filtering logic of interfaces for un/plug, testing under vm_tests.go is not always possible and takes way more time.

Reply:
> with no boilerplate whatsoever.

You can count the number of code lines that got removed once you just do the minimum as suggested. That is the boilerplate (e.g. wrapper function, tests for the wrapper function).

> Note that having these functions enable testing the filtering logic of interfaces for un/plug, testing under vm_tests.go is not always possible and takes way more time.

What interfaces to plug and unplug is not limited to the logic in these functions.

In the end it is all about balance. I think these helpers are not worth it.